security/logging/common.py

```diff
@@ -54,15 +63,35 @@ def set_slug(self, slug):
     def add_related_objects(self, *related_objects):
         self.related_objects |= set(related_objects)
 
+    @property
+    def extra_data(self):
+        return copy.deepcopy(self._extra_data)
```

Comment:
because extra data must be immutable

Reply:
and if you add extra data to model as a dict and change it here in the model it will be change too. It leads to problem where not changed data is not saved